src/components/radio/daikin-radio.ts

```diff
@@ -0,0 +1,218 @@
+    :host {
+      display: inline-block;
+    }
+  `;
+
+  private _handleClick(event: MouseEvent) {
+    if (this.readonly) {
+      event.preventDefault();
+    }
+  }
+
+  static formAssociated = true;
+
+  // define internals to let radio can be used in form
+  @property({ type: Object })
+  internals;
+
+  constructor() {
+    super();
+    this.internals = this.attachInternals();
+  }
+
+  private _updateFormValue() {
+    this.internals.setFormValue(this.checked ? this.value : null);
+  }
+
+  updated(changedProperties: Map<string, any>) {
+    if (changedProperties.has("checked")) {
+      this._updateFormValue();
+    }
+  }
+
+  private _handleChange(event: Event) {
+    const input = this.shadowRoot?.querySelector("input") as HTMLInputElement;
+    this.checked = input.checked;
+    this._updateFormValue();
+    const newEvent = new Event("change", event);
+    this.dispatchEvent(newEvent);
+  }
+
+  /**
+   * Specify the label text for check box
+   */
+  @property({ type: String })
+  label = "";
+
+  /**
+   * Specify the component size
+   */
+  @property({ type: String })
+  size: ComponentSizeProps = "small";
+
+  /**
+   * Specify the label position
+   * when `left` the label will be in left of radio, when `right` label will be in right of radio
+   */
+  @property({ type: String, attribute: "label-position" })
+  labelPosition: "left" | "right" = "right";
+
+  /**
+   * Specify whether the Radio should be disabled
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Specify whether the radio is read only
+   */
+  @property({ type: Boolean, reflect: true })
+  readonly = false;
+
+  /**
+   * Specify whether the radio is be checked
+   */
+  @property({ type: Boolean, reflect: true })
+  checked = false;
+
+  /**
+   * The form name.
+   */
+  @property({ type: String, reflect: true })
+  name = "";
+
+  /**
+   * The value.
+   */
+  @property({ type: String, reflect: true })
+  value = "";
+
+  /**
+   * Specify whether the Radio is in a error state
+   */
+  @property({ type: Boolean, reflect: true })
+  error = false;
+
+  render() {
+    // Specify the component size
+    const labelClassName = labelCN({ size: this.size });
+    const radioClassName = radioCN({ size: this.size });
+
+    const labelText = this.label
+      ? html`<span class="${labelClassName}">${this.label}</span>`
+      : html``;
+    const inputTag = html`<input
+      class="${radioClassName}"
+      type="radio"
+      name="${this.name}"
+      value="${this.value}"
+      aria-readonly=${this.readonly}
+      ?checked=${this.checked}
+      ?readonly=${this.readonly}
+      ?disabled=${this.disabled}
+      @click=${this._handleClick.bind(this)}
+      @change=${this._handleChange.bind(this)}
+    />`;
+    const inputArea =
+      this.labelPosition === "left"
+        ? html`${labelText}${inputTag}`
+        : html`${inputTag}${labelText}`;
+    return html`<label class="inline-flex gap-[8px] items-center"
```

Comment:
Might be better to fill parent container so that the user can extend label width

```suggestion
    return html`<label class="inline-flex w-full h-full gap-[8px] items-center"
```